vignettes/concepts_conventions.Rmd

```diff
@@ -0,0 +1,285 @@
+- rather, it is passed on to the derivation function which evaluates it in its own environment. `exprs()` is the plural version of 
+`expr()`, so it accepts multiple comma-separated items and returns a list of expressions. 
+
+```{r, eval = TRUE}
+library(rlang)
+
+adae <- data.frame(USUBJID = "XXX-1", AEDECOD = "HEADACHE")
+
+# Return the adae object
+adae
+
+# Return an expression
+expr(adae)
+```
+
+When used within the contest of an `{admiral}` derivation function, `expr()` and `exprs()`  allow the function to evaluate the 
+expressions in the context of the input dataset. As an example, `expr()` and `exprs()` allow users to pass variable names of datasets 
+to the function without wrapping them in quotation marks. 
+
+The expressions framework is powerful because users are able to intuitively "inject code" into `admiral` functions
+(through the function parameters) using very similar syntax as if they were writing open code, with the exception possibly being an
+outer `exprs()` wrapper.  For instance, in the `derive_vars_merged()` call below, the user is merging `adsl` with `ex` and is able
+to filter `ex` prior to the merge using an expression passed to the `filter_add` parameter. Because `filter_add` accepts expressions, 
+the user has full power to filter their dataset as they please. In the same vein, the user is able to create any new variables they 
+wish after the merge using the `new_vars` argument, to which they pass a list of expressions containing "standard" R code.
+
+``` {r, eval = FALSE}
+derive_vars_merged(
+  adsl,
+  dataset_add = ex,
+  filter_add = !is.na(EXENDTM),
+  by_vars = exprs(STUDYID, USUBJID),
+  new_vars = exprs(
+    TRTEDTM = EXENDTM,
+    TRTETMF = EXENTMF,
+    COMPTRT = if_else(!is.na(EXENDTM), "Y", "N")
+  ),
+  order = exprs(EXENDTM),
+  mode = "last"
+)
+```
+
+### Bang-Bang (`!!`) and Bang-Bang-Bang (`!!!`) {#unquoting}
+
+Sometimes you may want to construct an expression using other, pre-existing expressions. However, it's not immediately clear how
+to achieve this because expressions inherently pause evaluation of your code before it's executed:
+
+```{r, eval = TRUE}
+a <- expr(2)
+b <- expr(3)
+
+expr(a + b)
+# NOT 2 + 3
+```
+
+This is where `!!` (bang-bang) comes in: provided again by the `{rlang}` package, it  allows you to inject the contents of an 
+expression into another expression, meaning that by using `!!` you can modify the code inside an expression before R evaluates it.
+By using `!!` you are **unquoting** an expression, i.e. evaluating it before you pass it onwards.
+
+
+```{r, eval = TRUE}
+expr(!!a + !!b)
+```
+
+You can see an example of where `!!` comes in handy within `{admiral}` code in [Common Pitfall 1](#pitfall1), where the contents of an expression is unquoted so that it can be passed to `derive_vars_merged()`.
+
+`!!!` (bang-bang-bang) is the plural version of `!!` and can be used to unquote a list of expressions:
+
+```{r, eval = TRUE}
+exprs(!!!list(a, b))
```

Comment:
Do we use this anywhere for user-facing code?  Can we make example if so since we are talking about it.